### PR TITLE
[mimalloc] Update to 1.6.4

### LINKED ISSUE #12933: [mimalloc] update to v1.6.4
Library name: mimalloc

New version number: v.1.6.4

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix #12933

Update mimalloc to the latest version 1.6.4

All features are tested successfully in the following triplets:
- x86-windows
- x64-windows
- x64-windows-static

### CODE REVIEW

File: ports/mimalloc/fix-cmake.patch [hunk 1]
```diff
@@ -1,70 +1,54 @@
-diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (ras0219-msft) [inline ports/mimalloc/fix-cmake.patch:1]: Please change the line endings back to LF

File: ports/mimalloc/fix-cmake.patch [hunk 2]
```diff
@@ -1,70 +1,54 @@
...
++    install (
++      FILES $<TARGET_FILE_DIR:mimalloc>/mimalloc-redirect32.dll
++      DESTINATION ${CMAKE_INSTALL_BINDIR}
++   )
++  endif()	  
++endif()
+ 
+-  install(TARGETS mimalloc EXPORT mimalloc DESTINATION ${mi_install_dir} LIBRARY)
++  install(TARGETS mimalloc EXPORT mimalloc ARCHIVE DESTINATION lib RUNTIME DESTINATION bin LIBRARY DESTINATION lib NAMELINK_SKIP)
+   install(EXPORT mimalloc DESTINATION ${mi_install_dir}/cmake)
```
Comment (ras0219-msft) [inline ports/mimalloc/fix-cmake.patch:33]: This export statement needs to be moved after the `if (MI_BUILD_STATIC)`/`endif()` block below

File: ports/mimalloc/portfile.cmake [hunk 3]
```diff
@@ -28,6 +31,9 @@ vcpkg_configure_cmake(
         -DMI_USE_CXX=OFF
         -DMI_BUILD_TESTS=OFF
         ${FEATURE_OPTIONS}
+	OPTIONS
+	    -DKEYSTONE_BUILD_STATIC=${MI_BUILD_STATIC}
```
Comment (ras0219-msft) [inline ports/mimalloc/portfile.cmake:35]: These need to be `MI_BUILD_STATIC` (and same for shared)

Comment (ras0219-msft) [inline ports/mimalloc/portfile.cmake:35]: There is a tab instead of spaces here; please use spaces

File: ports/mimalloc/portfile.cmake [hunk 4]
```diff
@@ -28,6 +31,9 @@ vcpkg_configure_cmake(
         -DMI_USE_CXX=OFF
         -DMI_BUILD_TESTS=OFF
         ${FEATURE_OPTIONS}
+	OPTIONS
```
Comment (ras0219-msft) [inline ports/mimalloc/portfile.cmake:34]: This `OPTIONS` header is unneeded -- this is already an options section.

File: ports/mimalloc/portfile.cmake [hunk 5]
```diff
@@ -38,6 +44,7 @@ file(GLOB lib_directories RELATIVE ${CURRENT_PACKAGES_DIR}/lib "${CURRENT_PACKAG
 list(GET lib_directories 0 lib_install_dir)
 vcpkg_fixup_cmake_targets(CONFIG_PATH lib/${lib_install_dir}/cmake)
 
+
```
Comment (ras0219-msft) [inline ports/mimalloc/portfile.cmake:47]: Please remove this extra whitespace

File: ports/mimalloc/portfile.cmake [hunk 6]
```diff
@@ -56,6 +63,12 @@ file(REMOVE_RECURSE
     ${CURRENT_PACKAGES_DIR}/debug/share
     ${CURRENT_PACKAGES_DIR}/lib/${lib_install_dir}
 )
+if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
+    file(REMOVE_RECURSE 
+		${CURRENT_PACKAGES_DIR}/bin 
+		${CURRENT_PACKAGES_DIR}/debug/bin
+)
+endif()
```
Comment (ras0219-msft) [inline ports/mimalloc/portfile.cmake:71]:
```suggestion
```
These lines are not needed once the above cmake options are fixed.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

File: ports/mimalloc/portfile.cmake (hunk 6 above)
Comment (JonLiu1993) [reply ports/mimalloc/portfile.cmake:71]: Thank you for your guidance and correction, I will revise and submit again, thanks ！

**Comment by JonLiu1993** [comment]
Duplicated to [#13629 ](https://github.com/microsoft/vcpkg/pull/13629)

**Pull request closed without merging**